### ISSUE TITLE
Tooltips / glossary / help button for advanced concepts

### ISSUE THREAD

**Issue by mvdbeek** [galaxyproject/galaxy#12934]
@jmchilton added some nice tooltips in #12912, and I think we should/could take this a bit further and describe a few basic unix concepts/terms in more detail (especially since they appear in the user interface and could be daunting to new users), like
- [ ] Exit Code
- [ ] Stdout
- [ ] Stderr
- [ ] Command line
- [ ] Traceback

Maybe this could be based on the how we do translations, and we could have a little help icon if help exists for these terms (I don't love tooltips, they can get in the way ...).

In the same way we should probably also provide a little help for Galaxy specific terms, like
- [ ] History ID
- [ ] History Content ID
- [ ] Dataset Collection / History Dataset Collection Association
- [ ] History Dataset Association
- [ ] Dataset
- [ ] Invocation
- [ ] Datatype
- [ ] Dbkey
- [ ] Job states

I think this could be a cool little collaborative project.

**Comment by assuntad23** [comment]
## Functional Description:
Users can enable help mode on Galaxy to find and learn about features of the UI that are not very apparent at first glance. This help mode can be toggled on and off and will need to be manually added to any potential components that could benefit from the attention.  The component should be able to route to other parts of the Galaxy interface, link to GTN materials or other supplementary materials, pop-up additional context or help text for users, and know the job status of an item.  There will also exist a database table of all glossary items and only items that are “unlearned” will display. Users should be able to “learn” and thereby disable glossary items that are not relevant to them. 

## Context:
- Newer users are overwhelmed by the interface
- Hard to find certain areas of interest or know how to use them without connecting them to tutorials or training materials
- Complex components that have a lot of power, but are only usable by folks who know how to utilize that power

## Objective:
- Users can enable “Help Mode”
- Help Mode will highlight certain more confusing aspects of the UI and link users out to training materials or other supplementary materials

**Comment by mvdbeek** [comment]
> There will also exist a database table of all glossary items and only items that are “unlearned” will display. Users should be able to “learn” and thereby disable glossary items that are not relevant to them.

I personally think that might not be necessary for a first pass. My idea was that we should just have a way to provide users with a way to learn what a History Dataset or an Invocation is with a `?` icon that you can click on. And if the invocation heklp mentions History Datasets you should be able to click through to that glossary term. I was hoping for this to be integrated in the normal flow of the app, without a separate mode.

**Comment by assuntad23** [comment]
Out of UI/UX meeting: 
I'm going to work on a Figma design for the side panel/help text that populates when you hover over an item that has help text.  (Thanks @ElectronicBlueberry !) 
And a glow/highlight design that glows when you hover over it, and then you can click it for the popover with help text. (Thanks @dannon ) 
Then we can take a look at all three implementations alongside eachother and see if one stands out more than the others. 

If anyone has any other ideas for UI implementation, let me know, so I can try to present that too.

**Comment by assuntad23** [comment]
[Here's some Designs for the other two ideas: ](https://www.figma.com/file/SIq1FiG9hCxU94ijumYLtD/Interactive-Glossary?type=design&node-id=115-2&t=GIWvMwLDCI308Y5r-0)

Glow Design:

f![Screenshot from 2023-05-02 17-00-09](https://user-images.githubusercontent.com/26912553/235785239-accd1003-8eca-4c2b-bab1-07a02ca87168.png)


Panel (I put it in between the main screen and the History, but we could put it on the end too):

![Screenshot from 2023-05-02 17-02-07](https://user-images.githubusercontent.com/26912553/235785445-195f34b2-3147-4597-ac4d-6f91caafca45.png)


Both of these have seperate drawbacks. 
The panel (as discussed in UI/UX meeting) would not operate well with our heavy use of modals. The *not-so-simple* answer here is to get rid of modals. But I'm open to other thoughts, too. 
The highlight/glow implementation with popover (on click) would make it impossible to actually click the item. I suppose the answer for this one is to display the popover on hover instead.

**Comment by tcollins2011** [comment]
Visually I prefer the Glow design more than the Panel design. It could also be possible to have the glow design popover on hover over the target element. I was also wondering if it would make sense to possibly implement this information as a tour as well so that a new user is guided to each of the elements rather than just naturally discovering them.

**Comment by assuntad23** [comment]
> I was also wondering if it would make sense to possibly implement this information as a tour as well so that a new user is guided to each of the elements rather than just naturally discovering them.

I think adding a tour would be a nice follow-up, we can add that on later, for sure. Depending on how many different items/components we choose to highlight in this way, it might make sense to have multiple tours, broken down by "area".

**Comment by hexylena** [comment]
Agreed with @ksuderman, having a way to revisit or re-activate old "learned" items is really useful for cognitive accessibility concerns (looking at you, every video game that doesn't let me revisit the interface tutorial when I've been away from it for a few months.)

**Comment by assuntad23** [comment]
^ yes, @ksuderman and @hexylena  that was the idea that I was working towards. I also thought it might be useful to allow users to re-enable things that they'd previously dismissed in User Preferences. 

I also tend to quickly click that "yes, I know this thing" only to realize that I need to see it again.

**Comment by assuntad23** [comment]
Adding some voting here: 
🎉  for the "?" popover implementation
👀  for the glowing/highlight implementation
🚀  for the side panel implementation

if you already voted on the standup channel, no need to vote again here.

**Comment by ElectronicBlueberry** [comment]
In context of the UI/UX WG meeting: here's an example of how the info pane could alternatively be implemented:

![image](https://github.com/galaxyproject/galaxy/assets/44241786/98118ea8-7528-4f47-809e-8e1184ecaaa0)

(taken from ableton live)

**Comment by assuntad23** [comment]
@ElectronicBlueberry do you think a user would want to see this at the bottom of the tool panel or at the bottom of their history? That's why I added a whole new panel for it, because I couldn't decide which panel it should sit below.

**Comment by ElectronicBlueberry** [comment]
Tool Panel. The history requires more vertical screen space, due to the header, and height of the history items
